tests/test_jax_nonbonded.py

```diff
@@ -0,0 +1,151 @@
+import jax
+jax.config.update("jax_enable_x64", True)
+
+import numpy as onp
+from numpy.random import randn, rand, randint, seed
+seed(2021)
+
+from jax import numpy as np, value_and_grad, jit, vmap
+
+from jax.ops import index_update, index
+from timemachine.potentials.nonbonded import nonbonded_v3, nonbonded_v3_on_specific_pairs
+from timemachine.potentials.jax_utils import convert_to_4d, get_all_pairs_indices, get_group_group_indices
+
+from functools import partial
+from typing import Tuple, Callable
+
+Conf = Params = Box = ChargeMask = LJMask = LambdaPlaneIdxs = LambdaOffsetIdxs = np.array
+Lamb = Beta = Cutoff = Energy = float
+
+nonbonded_args = Conf, Params, Box, Lamb, ChargeMask, LJMask, Beta, Cutoff, LambdaPlaneIdxs, LambdaOffsetIdxs
+NonbondedArgs = Tuple[nonbonded_args]
+NonbondedFxn = Callable[[*nonbonded_args], Energy]
+
+
+def generate_random_inputs(n_atoms: int, dim: int = 3) -> NonbondedArgs:
+    # jittered
+    offsets = onp.array([onp.arange(n_atoms)] * dim).T
+    assert (offsets.shape == (n_atoms, dim))
+    for i in range(dim):
+        onp.random.shuffle(offsets[:, i])
+    conf = (0.5 * randn(n_atoms, dim) + offsets)
+    conf += conf.min()
+    params = rand(n_atoms, 3)
+    params[:, 0] -= np.mean(params[:, 0])
+
+    box = np.diag(0.5 + 0.5 * rand(3)) * 10 * n_atoms
```

Comment:
I would add a test using a box that actually triggers the PBC wrapping behavior. I think this creates a really large box most of the time such that the system is effectively aperiodic.

Reply:
Addressed in https://github.com/proteneer/timemachine/pull/453/commits/a28424d2bd60338587d7eedc92bf0573fdb2460a